orion/primitives/jsons/orion.primitives.tadgan.TadGAN.json

```diff
@@ -1,356 +1,373 @@
-                      "parameters": {
-                          "units": "dense_units"
-                      }
-                  },
-                  {
-                      "class": "keras.layers.advanced_activations.LeakyReLU",
-                      "parameters": {
-                          "alpha": 0.2
-                      }
-                  },
-                  {
-                      "class": "keras.layers.Dropout",
-                      "parameters": {
-                          "rate": 0.2
-                      }
-                  },
-                  {
-                      "class": "keras.layers.Dense",
-                      "parameters": {
-                          "units": "dense_units"
-                      }
-                  },
-                  {
-                      "class": "keras.layers.advanced_activations.LeakyReLU",
-                      "parameters": {
-                          "alpha": 0.2
-                      }
-                  },
-                  {
-                      "class": "keras.layers.Dropout",
-                      "parameters": {
-                          "rate": 0.2
-                      }
-                  },
-                  {
-                      "class": "keras.layers.Dense",
-                      "parameters": {
-                          "units": 1
-                      }
-                  }
-              ]
-          }
-      }
-  }
+    "name": "orion.primitives.tadgan.TadGAN",
+    "contributors": [
+        "Alexander Geiger <geigera@mit.edu>",
+        "Dongyu Liu <dongyu@mit.edu>",
+        "Lawrence Wong <lcwong@mit.edu>"
+    ],
+    "documentation": "",
+    "description": "time series GAN for anomaly detection",
+    "classifiers": {
+        "type": "estimator",
+        "subtype": "regressor"
+    },
+    "modalities": [],
+    "primitive": "orion.primitives.tadgan.TadGAN",
+    "fit": {
+        "method": "fit",
+        "args": [
+            {
+                "name": "X",
+                "type": "ndarray"
+            },
+            {
+                "name": "y",
+                "type": "ndarray"
+            }
+        ]
+    },
+    "produce": {
+        "method": "predict",
+        "args": [
+            {
+                "name": "X",
+                "type": "ndarray"
+            },
+            {
+                "name": "y",
+                "type": "ndarray"
+            }
+        ],
+        "output": [
+            {
+                "name": "y",
+                "type": "ndarray"
+            },
+            {
+                "name": "critic",
+                "type": "ndarray"
+            }
+        ]
+    },
+    "hyperparameters": {
+        "fixed": {
+            "optimizer": {
+                "type": "str",
+                "default": "tensorflow.keras.optimizers.Adam"
+            },
+            "input_shape": {
+                "type": "tuple",
+                "default": null
```

Comment:
null? If this variable is not used, it should be deleted. The same for other null variables in fixed section.

Reply:
It defaults to null so that `_augment_hyperparameters` could infer it if it is null.